src/index.cpp

```diff
@@ -289,73 +290,15 @@ void Index<T, TagT, LabelT>::save(const char *filename, bool compact_before_save
     {
         if (_filtered_index)
         {
-            if (_label_to_start_id.size() > 0)
-            {
-                std::ofstream medoid_writer(std::string(filename) + "_labels_to_medoids.txt");
-                if (medoid_writer.fail())
-                {
-                    throw diskann::ANNException(std::string("Failed to open file ") + filename, -1);
-                }
-                for (auto iter : _label_to_start_id)
-                {
-                    medoid_writer << iter.first << ", " << iter.second << std::endl;
-                }
-                medoid_writer.close();
-            }
-
-            if (_use_universal_label)
+            _filter_store->save_medoids(std::string(filename) + "_labels_to_medoids.txt");
```

Comment:
Is it better to have single save function in filterstore and internally it will use save_medoids , save_labels_map, save_universal_label and save_labels function

Reply:
We can indeed have that, but this gives us a more refined control. It's just a matter of choice.